src/safeposix/syscalls/net_calls.rs

```diff
@@ -1272,21 +1383,19 @@ impl Cage {
                         );
                     }
 
+                    //The destaddr's address family must match that of the
+                    //socket's address from 'fd'. Otherwise, a message can't be sent
                     if dest_addr.get_family() != sockhandle.domain as u16 {
                         return syscall_error(
                             Errno::EINVAL,
                             "sendto",
                             "An address with an invalid family for the given domain was specified",
                         );
                     }
-                    if (flags & !MSG_NOSIGNAL) != 0 {
-                        return syscall_error(
-                            Errno::EOPNOTSUPP,
-                            "sendto",
-                            "The flags are not understood!",
-                        );
-                    }
 
+                    //If sendto_syscall is used on a connection-mode socket, then
```

Comment:
how is this check different from the one below it?

Reply:
A case that distinguishes them is in the case that a UDP socket is connected. The first check would catch that but the second wouldn't. Thanks for pointing that out.

Reply:
i am confused, if its a connection mode socket, and destaddress is null, we wouldnt reach this line in execusion right?

Reply:
If the socket is in some CONNECTED mode, such as CONNECTED, CONNWRONLY, CONNRDONLY, then it continues, ignoring the destaddress that is null